components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/util/ActionBuilderUtil.java

```diff
@@ -0,0 +1,280 @@
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtClientException;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtException;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.ActionRule;
+import org.wso2.carbon.identity.action.management.model.Authentication;
+import org.wso2.carbon.identity.action.management.model.EndpointConfig;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionBasicResponse;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionModel;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionResponse;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionType;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionUpdateModel;
+import org.wso2.carbon.identity.api.server.action.management.v1.AuthenticationTypeResponse;
+import org.wso2.carbon.identity.api.server.action.management.v1.EndpointResponse;
+import org.wso2.carbon.identity.api.server.action.management.v1.Link;
+import org.wso2.carbon.identity.api.server.action.management.v1.builder.ActionBuilder;
+import org.wso2.carbon.identity.api.server.action.management.v1.builder.PreIssueAccessTokenActionBuilder;
+import org.wso2.carbon.identity.api.server.action.management.v1.builder.PreUpdatePasswordActionBuilder;
+import org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_EMPTY_ACTION_ENDPOINT_AUTHENTICATION_PROPERTIES;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_INVALID_ACTION_ENDPOINT_AUTHENTICATION_PROPERTIES;
+
+/**
+ * Utility class for Action Builder.
+ */
+public class ActionBuilderUtil {
+
+    /**
+     * Get ActionBuilder object based on the action type.
+     *
+     * @param actionType Action type.
+     * @return ActionBuilder object.
+     */
+    public static ActionBuilder getActionBuilder(Action.ActionTypes actionType) {
+
+        ActionBuilder actionBuilder = null;
+        switch (actionType) {
+            case PRE_ISSUE_ACCESS_TOKEN:
+                actionBuilder = new PreIssueAccessTokenActionBuilder();
+                break;
+            case PRE_UPDATE_PASSWORD:
+                actionBuilder = new PreUpdatePasswordActionBuilder();
+                break;
+            default:
+                break;
+        }
+
+        if (actionBuilder == null) {
+            throw ActionMgtEndpointUtil.handleException(Response.Status.BAD_REQUEST,
+                    ActionMgtEndpointConstants.ErrorMessage.ERROR_NO_ACTION_BUILDER_FOUND);
```

Comment:
What's the content of this message. Couldn't find it here. It's not good to respond as a builder is not found which is an internal detail. If that's what this do, it needs to be fixed

Reply:
Fixed